src/app/games/diam/DiamFailure.ts

```diff
@@ -0,0 +1,13 @@
+import { Localized } from 'src/app/utils/LocaleUtils';
+
+export class DiamFailure {
+    public static readonly NO_MORE_PIECES_OF_THIS_TYPE: Localized = () => $localize`You do not have any pieces of this type anymore.`;
+
+    public static readonly STACK_IS_FULL: Localized = () => $localize`You cannot play here: this space is already full.`;
+
+    public static readonly TARGET_STACK_TOO_HIGH: Localized = () => $localize`You cannot add more pieces to the target space, as it will contain more than 4 pieces.`;
+
+    public static readonly MUST_SHIFT_LEFT_OR_RIGHT: Localized = () => $localize`To perform a shift, you must move the pieces to the space either directly to the left or directly to the right.`;
```

Comment:
haha, clock-wise or anti-clockwise, perhaps :p ?
Nicer to read with right and left less accurate :D 
Just say to one of the two direct neighbors perhaps :D

Reply:
changé to "neighboring space", et changé les right/left en clockwise/anticlockwise